### PR TITLE
Add example about how to apply hotfix to dm cluster

### PR DESCRIPTION
**Pull request by sunzhaoyang**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@sunzhaoyang Please involve a technical review. Thanks!

**Review by lance6716 — APPROVED** [review]

**Comment by glkappe** [comment]
Please add a check to see if the hotfix binary has executable permissions
Improve the success rate of patch

image
image

**Review by AstroProfundis — APPROVED** [review]
LGTM

**Comment by AstroProfundis** [comment]
> maybe it is a new feature of tiup

It was added quite long ago in v1.3.1 (#1091)[https://github.com/pingcap/tiup/pull/1091]

File: tiup/tiup-component-dm-patch.md [hunk 1]
```diff
@@ -69,8 +70,145 @@ tiup dm patch <cluster-name> <package-path> [flags]
 - 数据类型：`BOOLEAN`
 - 该选项默认关闭，默认值为 `false`。在命令中添加该选项，并传入 `true` 值或不传值，均可开启此功能。
 
-## 输出
+## 在线应用 Hotfix 示例
```
Comment (ran-huang) [inline tiup/tiup-component-dm-patch.md:73]: 这一部分和 L25-L37 重复了。

要不把这个部分抽出来单独写一篇 DM 角度的文档吧。现在这篇是从 tiup 的角度写的命令快速 reference，不是一个 task-based 的文档。

File: tiup/tiup-component-dm-patch.md [hunk 2]
```diff
@@ -69,8 +70,145 @@ tiup dm patch <cluster-name> <package-path> [flags]
...
-tiup-dm 的执行日志。
+以下将在使用 TiUP 部署的 DM 环境中演示如何应用`v5.3.0-hotfix`到`v5.3.0`集群，其他部署方式可能需要调整部分操作。
+
+> **注意：**
+>
+> hotfix 仅用于紧急修复，其日常维护较为复杂，建议在正式版本发布后及时升级。
+
+### 准备工作
+
+在开始应用 Hotfix 之前，请准备好 hotfix 补丁包 `dm-linux-amd64.tar.gz`。并确认当前环境 DM 软件版本：
```
[the remark this answers is not in the document]
Comment (sunzhaoyang) [reply tiup/tiup-component-dm-patch.md:83]: hotfix 都是人肉获取....

[the remark this answers is not in the document]
Comment (ran-huang) [reply tiup/tiup-component-dm-patch.md:83]: 意思是没有公开的下载包，而是 PingCAP 技术支持这边直接给到客户？

[the remark this answers is not in the document]
Comment (sunzhaoyang) [reply tiup/tiup-component-dm-patch.md:83]: Yes

Comment (shichun-0415) [inline tiup/tiup-component-dm-patch.md:83]: 确认 DM 版本的标准是什么？比如不得低于 5.X。
```suggestion
在开始应用 Hotfix 之前，请准备好 Hotfix 补丁包 `dm-linux-amd64.tar.gz`。并确认当前环境 DM 软件版本：
```

File: tiup/tiup-component-dm-patch.md [hunk 3]
```diff
@@ -69,8 +70,145 @@ tiup dm patch <cluster-name> <package-path> [flags]
...
+
+```shell
+/home/tidb/dm/deploy/dm-master-8261/bin/dm-master/dm-master -V
+```
+
+输出示例
+
+```
+Release Version: v5.3.0
```
Comment (shichun-0415) [inline tiup/tiup-component-dm-patch.md:92]: 是否构建最新的版本？

File: tiup/tiup-component-dm-patch.md [hunk 4]
```diff
@@ -69,8 +70,145 @@ tiup dm patch <cluster-name> <package-path> [flags]
...
+### 制作 Patch 补丁包并应用到 DM 集群
+
+1. 准备当前环境版本的 DM 软件包
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    mkdir -p /tmp/package
+    tar -zxvf /root/.tiup/storage/dm/packages/dm-master-v5.3.0-linux-amd64.tar.gz -C /tmp/package/
+    tar -zxvf /root/.tiup/storage/dm/packages/dm-worker-v5.3.0-linux-amd64.tar.gz -C /tmp/package/
```
Comment (shichun-0415) [inline tiup/tiup-component-dm-patch.md:108]: 确认版本是否更新

File: tiup/tiup-component-dm-patch.md [hunk 5]
```diff
@@ -69,8 +70,145 @@ tiup dm patch <cluster-name> <package-path> [flags]
...
+    ```
+
+2. 替换新的二进制文件
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    # 解压 hotfix 压缩包并替换
```
Comment (shichun-0415) [inline tiup/tiup-component-dm-patch.md:116]: 替换，是谁替换谁？
```suggestion
    # 解压 Hotfix 压缩包并替换
```

File: tiup/tiup-component-dm-patch.md [hunk 6]
```diff
@@ -69,8 +70,145 @@ tiup dm patch <cluster-name> <package-path> [flags]
...
+
+    # 重新打包
+    # 注意，其他部署方式可能有所不同
+    cd /tmp/package/ && tar -czvf dm-master-hotfix-linux-amd64.tar.gz dm-master/
+    cd /tmp/package/ && tar -czvf dm-worker-hotfix-linux-amd64.tar.gz dm-worker/
+    ```
+
+3. 应用补丁
+
+    查询当前集群状态, 以集群名称`dm-test`为示例
```
Comment (shichun-0415) [inline tiup/tiup-component-dm-patch.md:129]:
```suggestion
    查询当前集群状态, 以名称为 `dm-test` 的集群为例
```

File: tiup/tiup-component-dm-patch.md [hunk 7]
```diff
@@ -69,8 +70,145 @@ tiup dm patch <cluster-name> <package-path> [flags]
...
+    ```shell
+    tiup dm display dm-test
+    ```
+
+    输出示例
+
+    ```
+    Cluster type:       dm
+    Cluster name:       dm-test
+    Cluster version:    v5.3.0
```
Comment (shichun-0415) [inline tiup/tiup-component-dm-patch.md:142]: 版本号是否更新

File: tiup/tiup-component-dm-patch.md [hunk 8]
```diff
@@ -69,8 +70,145 @@ tiup dm patch <cluster-name> <package-path> [flags]
...
+    --                  ----                 ----           -----      -------       ------     --------                              ----------
+    172.16.100.21:9093  alertmanager         172.16.100.21  9093/9094  linux/x86_64  Up         /home/tidb/dm/data/alertmanager-9093  /home/tidb/dm/deploy/alertmanager-9093
+    172.16.100.21:8261  dm-master            172.16.100.21  8261/8291  linux/x86_64  Healthy|L  /home/tidb/dm/data/dm-master-8261     /home/tidb/dm/deploy/dm-master-8261
+    172.16.100.21:8262  dm-worker            172.16.100.21  8262       linux/x86_64  Free       /home/tidb/dm/data/dm-worker-8262     /home/tidb/dm/deploy/dm-worker-8262
+    172.16.100.21:3000  grafana              172.16.100.21  3000       linux/x86_64  Up         -                                     /home/tidb/dm/deploy/grafana-3000
+    172.16.100.21:9090  prometheus           172.16.100.21  9090       linux/x86_64  Up         /home/tidb/dm/data/prometheus-9090    /home/tidb/dm/deploy/prometheus-9090
+    Total nodes: 5
+    ```
+
+    指定节点或指定角色应用补丁，若`-R`和`-N`同时使用将会取其交集
```
Comment (shichun-0415) [inline tiup/tiup-component-dm-patch.md:155]:
```suggestion
    将补丁应用到指定节点或指定角色，若 `-R` 和 `-N` 同时使用，将会取其交集
```

File: tiup/tiup-component-dm-patch.md [hunk 9]
```diff
@@ -69,8 +70,145 @@ tiup dm patch <cluster-name> <package-path> [flags]
...
+
+    {{< copyable "shell-regular" >}}
+
+    ```
+    # 为指定节点应用 patch
```
Comment (shichun-0415) [inline tiup/tiup-component-dm-patch.md:160]:
```suggestion
    # 为指定节点应用补丁
```

File: tiup/tiup-component-dm-patch.md [hunk 10]
```diff
@@ -69,8 +70,145 @@ tiup dm patch <cluster-name> <package-path> [flags]
...
+    tiup dm patch dm-test dm-master-hotfix-linux-amd64.tar.gz -N 172.16.100.21:8261
+    tiup dm patch dm-test dm-worker-hotfix-linux-amd64.tar.gz -N 172.16.100.21:8262
+
+    # 或者根据角色应用 patch
```
Comment (shichun-0415) [inline tiup/tiup-component-dm-patch.md:164]:
```suggestion
    # 为指定角色应用补丁
```

File: tiup/tiup-component-dm-patch.md [hunk 11]
```diff
@@ -69,8 +70,145 @@ tiup dm patch <cluster-name> <package-path> [flags]
...
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    /home/tidb/dm/deploy/dm-master-8261/bin/dm-master/dm-master -V
+    ```
+
+    输出示例
+
+    ```
+    Release Version: v5.3.0-20211230
```
Comment (shichun-0415) [inline tiup/tiup-component-dm-patch.md:180]: 确认版本和时间是否更新

File: tiup/tiup-component-dm-patch.md [hunk 12]
```diff
@@ -69,8 +70,145 @@ tiup dm patch <cluster-name> <package-path> [flags]
...
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    tiup dm display dm-test
+    ```
+
+    输出示例
+
+    ```
+    Starting component `dm`: /root/.tiup/components/dm/v1.8.1/tiup-dm display dm-test
```
Comment (shichun-0415) [inline tiup/tiup-component-dm-patch.md:198]: 1.8.1 是 tiup 版本么？需要更新么？

File: tiup/tiup-component-dm-patch.md [hunk 13]
```diff
@@ -69,8 +70,145 @@ tiup dm patch <cluster-name> <package-path> [flags]
...
+    Cluster type:       dm
+    Cluster name:       dm-test
+    Cluster version:    v5.3.0
```
Comment (shichun-0415) [inline tiup/tiup-component-dm-patch.md:201]: 版本号是否更新

File: tiup/tiup-component-dm-patch.md (hunk 3 above)
Comment (sunzhaoyang) [reply tiup/tiup-component-dm-patch.md:92]: No. hotfix 都是针对某个特性版本的，所以这里只是示例，并不需要新的版本

**Review by shichun-0415 — APPROVED** [review]
LGTM

File: tiup/tiup-component-dm-patch.md [hunk 14]
```diff
@@ -69,8 +70,145 @@ tiup dm patch <cluster-name> <package-path> [flags]
...
+
+    # 重新打包
+    # 注意，其他部署方式可能有所不同
+    cd /tmp/package/ && tar -czvf dm-master-hotfix-linux-amd64.tar.gz dm-master/
+    cd /tmp/package/ && tar -czvf dm-worker-hotfix-linux-amd64.tar.gz dm-worker/
+    ```
+
+3. 应用补丁
+
+    查询当前集群状态, 以名称为 `dm-test` 的集群为例
```
Comment (ran-huang) [inline tiup/tiup-component-dm-patch.md:129]:
```suggestion
    查询当前集群状态，以名称为 `dm-test` 的集群为例：
```

**Review by ran-huang — APPROVED** [review]

**Pull request merged**